schema/deploy/mutations/create_form_change.sql

```diff
@@ -0,0 +1,56 @@
+-- Deploy cif:mutations/create_form_change to pg
+
+begin;
+
+create function cif.create_form_change(
+  operation cif.form_change_operation,
+  json_schema_name varchar(1000),
+  form_data_schema_name varchar(1000),
+  form_data_table_name varchar(1000),
+  change_reason varchar(10000),
+  new_form_data jsonb default null,
+  form_data_record_id integer default null,
+  project_revision_id integer default null,
+  change_status varchar(1000) default 'pending',
+  validation_errors jsonb default '[]'
+) returns cif.form_change as $create_form_change$
+  insert into cif.form_change(
+    new_form_data,
+    operation,
+    form_data_schema_name,
+    form_data_table_name,
+    form_data_record_id,
+    project_revision_id,
+    change_status,
+    change_reason,
+    json_schema_name,
+    validation_errors
+  ) values (
+    coalesce(new_form_data, (
+      select new_form_data
+      from cif.form_change
+      where form_data_record_id = create_form_change.form_data_record_id
+      and form_data_schema_name = create_form_change.form_data_schema_name
+      and form_data_table_name = create_form_change.form_data_table_name
+      and change_status = 'committed'
+      order by updated_at desc, id desc
+      limit 1
+    )),
```

Comment:
Do we want to merge the found `new_form_data` in the previous form_change record, with what was passed in? In case there are hidden fields like a project_id that were only inserted in the json on the form_change creation, like [here](https://github.com/bcgov/cas-cif/blob/43b25015d77b9fa8dacc549807a3d900b4c2a0dc/schema/deploy/mutations/create_project.sql#L46).
I'm thinking something like
```psql
coalesce((select ...)::jsonb, {}::jsonb) || coalesce(new_form_data::jsonb, {}::jsonb)
```

Reply:
something like the project id would be copied over, so I'm not sure what we'd need to merge here